datadog/resource_datadog_dashboard.go

```diff
@@ -4601,6 +4852,158 @@ func getTimeseriesRequestSchema() map[string]*schema.Schema {
+		terraformSearch := terraformSearches[0].(map[string]interface{})
+		eventQuery.Search = datadogV1.NewTimeSeriesFormulaAndFunctionEventQueryDefinitionSearch(terraformSearch["query"].(string))
+	}
+
+	// GroupBy
+	if terraformGroupBys, ok := data["group_by"].([]interface{}); ok && len(terraformGroupBys) > 0 {
+		datadogGroupBys := make([]datadogV1.TimeSeriesFormulaAndFunctionEventQueryDefinitionGroupBy, len(terraformGroupBys))
+		for i, g := range terraformGroupBys {
+			groupBy := g.(map[string]interface{})
+
+			// Facet
+			datadogGroupBy := datadogV1.NewTimeSeriesFormulaAndFunctionEventQueryDefinitionGroupBy(groupBy["facet"].(string))
+
+			// Limit
+			if v, ok := groupBy["limit"].(int); ok && v != 0 {
+				datadogGroupBy.SetLimit(int64(v))
+			}
+
+			// Sort
+			if v, ok := groupBy["sort"].([]interface{}); ok && len(v) > 0 {
+				if v, ok := v[0].(map[string]interface{}); ok && len(v) > 0 {
+					sortMap := &datadogV1.TimeSeriesFormulaAndFunctionEventQueryDefinitionSort{}
+					if aggr, ok := v["aggregation"].(string); ok && len(aggr) > 0 {
+						aggregation := datadogV1.FormulaAndFunctionEventAggregation(v["aggregation"].(string))
+						sortMap.SetAggregation(aggregation)
+					}
+					if order, ok := v["order"].(string); ok && len(order) > 0 {
+						eventSort := datadogV1.QuerySortOrder(order)
+						sortMap.SetOrder(eventSort)
+					}
+					if metric, ok := v["metric"].(string); ok && len(metric) > 0 {
+						sortMap.SetMetric(metric)
+					}
+					datadogGroupBy.SetSort(*sortMap)
+				}
+			}
+
+			datadogGroupBys[i] = *datadogGroupBy
+		}
+		eventQuery.SetGroupBy(datadogGroupBys)
+	}
+
+	return datadogV1.TimeSeriesFormulaAndFunctionEventQueryDefinitionAsFormulaAndFunctionQueryDefinition(eventQuery)
+}
+
+func buildDatadogMetricQuery(data map[string]interface{}) datadogV1.FormulaAndFunctionQueryDefinition {
+	dataSource := datadogV1.FormulaAndFunctionMetricDataSource("metrics")
+	metricQuery := datadogV1.NewTimeSeriesFormulaAndFunctionMetricQueryDefinition(dataSource, data["query"].(string))
+	if v, ok := data["name"].(string); ok && len(v) != 0 {
+		metricQuery.SetName(v)
+	}
+	if v, ok := data["aggregator"].(string); ok && len(v) != 0 {
+		aggregator := datadogV1.FormulaAndFunctionMetricAggregation(data["aggregator"].(string))
+		metricQuery.SetAggregator(aggregator)
+	}
+
+	return datadogV1.TimeSeriesFormulaAndFunctionMetricQueryDefinitionAsFormulaAndFunctionQueryDefinition(metricQuery)
+}
+
+func buildDatadogFormulaAndFunctionProcessQuery(data map[string]interface{}) datadogV1.FormulaAndFunctionQueryDefinition {
+	dataSource := datadogV1.FormulaAndFunctionProcessQueryDataSource(data["data_source"].(string))
+	processQuery := datadogV1.NewTimeSeriesFormulaAndFunctionProcessQueryDefinition(dataSource, data["metric"].(string))
+
+	// Text Filter
+	if v, ok := data["text_filter"].(string); ok && len(v) != 0 {
+		processQuery.SetTextFilter(v)
+	}
+
+	terraformFilters := data["tag_filters"].([]interface{})
```

Comment:
Since this is optional, we may have to check its `ok` before we do the type assert. Can you check if the code panics if you create a process query without `tag_filters` set?

Reply:
No crashes when `tag_filters` are not set. 👌